src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -144,7 +140,78 @@ private void PollForTracingCommand(object state)
+        {
+            // Read the config file in once call.
+            byte[] configContents = File.ReadAllBytes(configFilePath);
+
+            // Convert the contents to a string.
+            string strConfigContents = Encoding.UTF8.GetString(configContents);
+
+            // Read all of the config options.
+            string outputPath = null;
+            string strProviderConfig = null;
+            string strCircularMB = null;
+
+            // Split the configuration entries by line.
+            string[] configEntries = strConfigContents.Split(ConfigFileLineDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            foreach (string configEntry in configEntries)
+            {
+                //`Split the key and value by '='.
+                string[] entryComponents = configEntry.Split(ConfigEntryDelimiter);
+                if(entryComponents.Length == 2)
+                {
+                    string key = entryComponents[0];
+                    if (key.Equals(ConfigKey_Providers))
+                    {
+                        strProviderConfig = entryComponents[1];
+                    }
+                    else if (key.Equals(ConfigKey_OutputPath))
+                    {
+                        outputPath = entryComponents[1];
+                    }
+                    else if(key.Equals(ConfigKey_CircularMB))
+                    {
+                        strCircularMB = entryComponents[1];
+                    }
+                }
+            }
+
+            // Ensure that the output path is set.
+            if (string.IsNullOrEmpty(outputPath))
+            {
+                throw new ArgumentNullException(nameof(outputPath));
+            }
+
+            // Build the full path to the trace file.
+            string traceFileName = Assembly.GetEntryAssembly().GetName().Name + "." + Win32Native.GetCurrentProcessId() + NetPerfFileExtension;
+            string outputFile = Path.Combine(outputPath, traceFileName);
+
+            // Get the circular buffer size.
+            uint circularMB = DefaultCircularBufferMB;
+            if(!string.IsNullOrEmpty(strCircularMB))
+            {
+                circularMB = Convert.ToUInt32(strCircularMB);
+            }
+
+            // Initialize a new configuration object.
+            EventPipeConfiguration config = new EventPipeConfiguration(outputFile, circularMB);
+
+            // Set the provider configuration if specified.
+            if (!string.IsNullOrEmpty(strProviderConfig))
+            {
+                SetProviderConfiguration(strProviderConfig, config);
+            }
+            else
+            {
+                // If the provider configuration isn't specified, use the default.
+                config.EnableProviderRange(DefaultProviderConfiguration);
+            }
+
+            return config;
+        }
+
+        private static EventPipeConfiguration BuildConfigFromEnvironment()
```

Comment:
I am wondering if we should bother at all with environment variable control as we currently have it.   Here are three alternatives.
1. No environment variables, simply probe for the file on startup. 
2. Have one envionrment variable - COMPLUS_EventPipeConfigFile which indicates where the config file lives.  
3.  Have one environment variable COMPLUS_EventPipeConfiguration, which is the contents of the configuration file (and the file can have one more thing in it, which is the name of the configuration file itself (in our case AppBase\EventPipeConfig).    

All of these simplify things (more code reuse, or just plain simpler).   All allow for profiling at startup, but the first requires that you have access to the app, and requires modifying files where you may not wish to.  The first also requires another file touch (but given that we probed for *.json files, I would expect that to not be a problem). 

Thoughts?